### DISCUSSION TITLE
Welcome to self-rewarding-lm-pytorch Discussions!

### DISCUSSION THREAD

**Discussion by lucidrains** [lucidrains/self-rewarding-lm-pytorch#2] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by mridulBanik112** [comment]
Thanks for doing the amazing work.
Can you please add in the readme or answer me how to run this code?
Thanks

**Reply by lucidrains** [reply]
all my repos, if you see `(wip)` it isn't done.

**Comment by JiHa-Kim** [comment]
Epic. I hope the results are truly working, then we can finally self-improve the model until AGI.

**Reply by lucidrains** [reply]
nah.. this work won't bring about AGI.. but it has some interesting ideas.

**Comment by nivibilla** [comment]
Hi @lucidrains ,

Thank you for your amazing work! I'm a Machine Learning Scientist at a travel company. Also I like to play around with llms in my free time.

This repo is very exciting if what Meta claims are true. However one suggestion I would like to make it possible is to allow users to train with LoRA. For GPU poor users like me it would allow to train these large models and I'm curious to see if training an adapter would still give the same performance as full finetuning.

Thanks!

**Reply by lucidrains** [reply]
sounds good, as a ML scientist, welcome you to jump in and participate with a PR

**Reply by nivibilla** [reply]
The authors of spin released their implementation.
https://github.com/uclaml/SPIN

It's not as clean but its built of HF transformers so has peft support

**Comment by xiechengmude** [comment]
any quick start scripts  for learning? thanks!

**Reply by lucidrains** [reply]
as I said above, `wip` flag means it isn't done.
